common/lib/capa/capa/capa_problem.py

```diff
@@ -950,21 +950,26 @@ def response_a11y_data(self, response, inputfields, responsetype_id, problem_dat
 
                 label = inputfields[0].attrib['label']
                 # Get first <p> tag before responsetype, this <p> contains the question text.
-                p_tag = response.xpath('preceding-sibling::p[1]')
+                p_tag_list = response.xpath('preceding-sibling::p[1]')
+                try:
+                    p_tag = p_tag_list[0]
+                except IndexError:
+                    p_tag = None
```

Comment:
@ehteshamkafeel Instead of `try...catch`, we can only use `p_tag = p_tag_list and p_tag_list[0]`. so the code looks like below

```Python
p_tag_list = response.xpath('preceding-sibling::p[1]')
p_tag = p_tag_list and p_tag_list[0]
if p_tag and len(p_tag.findall('*')) == 0 and p_tag.text:
.
.
.
```